activemodel/lib/active_model/error.rb

```diff
@@ -77,9 +77,12 @@ def self.generate_message(attribute, type, base, options) # :nodoc:
         attribute = attribute.to_s.remove(/\[\d+\]/)
 
         defaults = base.class.lookup_ancestors.flat_map do |klass|
-          [ :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{attribute}.#{type}",
-            :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.#{type}" ]
+          [ :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{attribute}.#{type}.message",
```

Comment:
It would be a simpler change for existing projects if the `message` lookup chain stays as it is. What do you think about a lookup chain for custom messages that adds `custom_messages` before the `type` instead of after it, like this? 🤔 
```
activemodel.errors.models.person.attributes.name.custom_messages.too_short
activemodel.errors.models.person.custom_messages.too_short
activemodel.errors.messages.custom_messages.too_short
activemodel.errors.attributes.name.custom_messages.too_short
errors.custom_messages.too_short
```

Reply:
I do remember this from your implementation but felt it a bit more coherent to nest specific keys underneath the error type. 

```yaml
en:
  errors:
    blank:
      message: can't be blank
      custom_message: You missed this one!
``` 

Vs

```yaml
en:
  errors:
    blank: can't be blank
    custom_message: 
      blank: You missed this one!
``` 

I feel this is a bit more tidy but I do see your point though. That's why I included the previous lookup path on the chain. 

@p8 any opinions on this one? Not sure how Rails considers potential impact on end users for changes like these.